server/scripts/generate_batch_file.py

```diff
@@ -1,105 +1,136 @@
-    with open(f"{GENERATE_BATCH_SCRIPT_PATH}{database}/test_{database}.json",'r') as file:
-        json_file=json.loads(file.read())
-        file.close()
+    total_estimated_cost = 0.0 
     
-    prompts=[]
-
-    if not response.status_code==APIStatusCode.SUCCESS.value:
-        print(response.json())
-        exit() 
-
-    # iterating over all NLP questions in each database
-    for item in tqdm(json_file,desc=f'Generating prompts for {database}'):
-
-        payload={'prompt_type':PROMPT_TYPE,'shots':NUM_SHOTS,'question':item['question']}
-        response=requests.post(PROMPT_GENERATE_ENDPOINT,json=payload)
-
-        if not response.status_code==APIStatusCode.SUCCESS.value:
-            print(response.json())
-            exit()
-         
-        prompts.append({
-                "custom_id": f"request-{item['question_id']}",
-                "method": "POST",
-                "url": "/v1/chat/completions",
-                "body": {
-                    "model": MODEL,
-                    "messages": [
-                        {"role": "system", "content": response.json()['generated_prompt']},
-                    ],
-                    "max_tokens": MAX_TOKENS,
-                    "temperature": TEMPERATURE
-                }
-            })
-
-    os.makedirs(f"{GENERATE_BATCH_SCRIPT_PATH}{database}{BATCH_DIR_SUFFIX}" ,exist_ok=True)
-    with open(f"{GENERATE_BATCH_SCRIPT_PATH}{database}{BATCH_DIR_SUFFIX}{BATCHINPUT_FILE_PREFIX}_{database}.jsonl",'w') as file:
-        for item in prompts:
-            json.dump(item, file)
-            file.write('\n')
-        file.close()
-
-
-
-print("BATCH INPUT FILES CREATED")
-# exit()
-
-
-openAI_client=OpenAI(api_key=OPENAI_API_KEY)
-uploaded_files=[]
-batch_jobs=[]
-batch_jobs_dict={}
-
-# for each DB upload the batch input file and create a batch job
-for i in tqdm(directories,desc='Uploading and Creating Batch Jobs'):
-
-    # upload batch input file
-    with open(f"{GENERATE_BATCH_SCRIPT_PATH}{i}{BATCH_DIR_SUFFIX}{BATCHINPUT_FILE_PREFIX}_{i}.jsonl",'rb') as file:
-        uploaded_file=openAI_client.files.create(file=file, purpose='batch')
-        file.close()
-    uploaded_files.append(uploaded_file.id)
-
-    # create batch job
-    batch = openAI_client.batches.create(
-                input_file_id=uploaded_file.id,
-                endpoint="/v1/chat/completions",
-                completion_window="24h",
+    metadata = {
+        "batch_info": {
+            "dataset_path": dataset_dir,
+            "candidates": {
+                str(key): value for key, value in prompt_types_with_shots.items()
+            },
+            "model": model.value,
+            "temperature": temperature,
+            "max_tokens": max_tokens,
+            "total_estimated_cost": 0.0,
+            "overall_status": None
+        },
+        "databases": {}
+    }
+
+
+    for database in tqdm(databases[:2], desc="Creating and running batch input files"):
```

Comment:
This shouldn't be indexed on [:2]. But not a big enough problem to warrant a separate commit, if another commit is done then maybe fix this.